Yoast/wordpress-seo
# Add notification post types with has archive string

## Summary

This PR can be summarized in the following changelog entry:

* Shows a notification with the message that users should check their post type archive settings when these settings are possibly disappeared in our previous version.

## Relevant technical choices:

* Created a method `Yoast_Notification_Center->remove_notification_by_id` that is a wrapper for getting a notification by id and if something found remove that notification. These logic was used multiple times. I've chosen for this `Tell don't ask principle` because it is the responsibility of the notification center.
* I've use `@codeCoverageIgnore` for a couple of wrapper methods because these methods have dependencies.

## Test instructions

This PR can be tested by following these steps:

* Have a couple of post types with `has_archive` being set as a string (to override archive slug). This can be done with the custom post type ui plugin: https://nl.wordpress.org/plugins/custom-post-type-ui/
* Set for one custom post type some templates in Yoast SEO - Search Appearence - Content types.
* Check if the notification is triggered.

## Quality assurance


Fixes #10382
